contracts/base/strategies/masterchef-base/HermesStrategyFullBuyback.sol

```diff
@@ -0,0 +1,150 @@
+  uint256 public poolID;
+
+  /// @notice Contract constructor using on strategy implementation
+  /// @dev The implementation should check each parameter
+  /// @param _controller Controller address
+  /// @param _underlying Underlying token address
+  /// @param _vault SmartVault address that will provide liquidity
+  /// @param __rewardTokens Reward tokens that the strategy will farm
+  /// @param _pool Hermes MasterChef pool address
+  /// @param _poolID Pool id
+  constructor(
+    address _controller,
+    address _underlying,
+    address _vault,
+    address[] memory __rewardTokens,
+    address _pool,
+    uint256 _poolID
+  ) StrategyBase(_controller, _underlying, _vault, __rewardTokens, _BUY_BACK_RATIO) {
+    require(_pool != address(0), "zero address pool");
+    pool = _pool;
+    poolID = _poolID;
+
+    (IERC20 lpToken,,,,,) = IIrisMasterChef(_pool).poolInfo(_poolID);
+    require(address(lpToken) == _underlyingToken, "wrong underlying");
+  }
+
+  // ************* VIEWS *******************
+
+  /// @notice Strategy balance in the Hermes MasterChef pool
+  /// @return bal Balance amount in underlying tokens
+  function rewardPoolBalance() public override view returns (uint256 bal) {
+    (bal,) = IIrisMasterChef(pool).userInfo(poolID, address(this));
+  }
+
+  /// @notice Return approximately amount of reward tokens ready to claim in Hermes MasterChef pool
+  /// @dev Don't use it in any internal logic, only for statistical purposes
+  /// @return Array with amounts ready to claim
+  function readyToClaim() external view override returns (uint256[] memory) {
+    uint256[] memory toClaim = new uint256[](1);
+    toClaim[0] = IIrisMasterChef(pool).pendingIris(poolID, address(this));
+    return toClaim;
+  }
+
+  /// @notice TVL of the underlying in the Hermes MasterChef pool
+  /// @dev Only for statistic
+  /// @return Pool TVL
+  function poolTotalAmount() external view override returns (uint256) {
+    return IERC20(_underlyingToken).balanceOf(pool);
+  }
+
+  /// @notice Calculate approximately weekly reward amounts for HERMES
+  /// @dev Don't use it in any internal logic, only for statistical purposes
+  /// @return Weekly reward amount of HERMES
+  function computeWeeklyPoolReward() public view returns (uint256) {
+    (, uint256 allocPoint, uint256 lastRewardBlock,,,) = IIrisMasterChef(pool).poolInfo(poolID);
+    uint256 time = block.number - lastRewardBlock;
+    uint256 irisPerBlock = IIrisMasterChef(pool).irisPerBlock();
+    uint256 totalAllocPoint = IIrisMasterChef(pool).totalAllocPoint();
+    uint256 sushiReward = time.mul(irisPerBlock).mul(allocPoint).div(totalAllocPoint);
+    uint256 averageBlockTime = 5;
+    return sushiReward * (1 weeks * 1e18 / time / averageBlockTime) / 1e18;
+  }
+
+  /// @notice Stubbed to zero
+  /// @return [0]
+  function poolWeeklyRewardsAmount() external pure override returns (uint256[] memory) {
+    uint256[] memory rewards = new uint256[](1);
+    rewards[0] = 0;
```

Comment:
use computeWeeklyPoolReward()
you can move this logic to this function coz only 1 reward token